src/server/session.ts

```diff
@@ -286,14 +358,33 @@ namespace ts.server {
             this.byteLength = opts.byteLength;
             this.hrtime = opts.hrtime;
             this.logger = opts.logger;
-            this.canUseEvents = opts.canUseEvents;
+            this.eventPort = opts.eventPort;
+            this.canUseEvents = opts.canUseEvents || !!opts.event;
 
             const { throttleWaitMilliseconds } = opts;
 
+            if (opts.event) {
+                this.event = opts.event;
+            }
+            else if (this.eventPort && this.canUseEvents) {
+                const eventSender = new SocketEventSender(this.host, this.logger, this.eventPort);

```

Comment:
I've retained the initialization of the socket here for backwards compatibility with users initializing `Session` with the `eventPort` argument. This way, the typings installer will be able to send events through the socket by calling `event` on the Session.

Reply:
Oops, realized that I actually added the `eventPort` argument in an earlier commit of this PR, so this is unnecessary. Will fix.